lib/active_merchant/billing/gateways/decidir.rb

```diff
@@ -266,6 +266,11 @@ def error_code_from(response)
           error_code ||= error['type']
         elsif response['error_type']
           error_code = response['error_type'] if response['validation_errors']
+        elsif error = response.dig('error')
+          validation_errors = error.dig('validation_errors', 0)
+          code = validation_errors['code'] if validation_errors && validation_errors['code']
+          param = validation_errors['param'] if validation_errors && validation_errors['param']
+          error_code = "#{error['error_type']} | #{code} | #{param}" if error['error_type']
```

Comment:
It looks like we won't return information about the code or the param if there's no `'error_type'` -- it's possible that this is what we want, but I wanted to double check.

Reply:
Hard to say, being that this is the only example of this type of response from Decidir, but I think we want to just resort to the last part which will return the `processor_error` incase the `error_code` is `nil`. Since this is such a unique case, I think the majority of error's will be handled by the first 2 conditionals.

Reply:
The other way I was thinking of doing it, was to just grab that `error_type` and `validation_errors`, which would have ended up looking a little sloppy: 

`invalid_request_error | [{"code"=>"invalid_param", "param"=>"payment_type"}]`

I feel like that might end up being more difficult for the users to parse, downstream.

Reply:
I'm okay with leaving it how it is -- I'm sure someone will let us know if they're still not getting useful info 😅